[jit] Make torch.jit.Attribute work with PYTORCH_ENABLED=0

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23851 [jit] Make torch.jit.Attribute work with PYTORCH_ENABLED=0**

Differential Revision: [D16840394](https://our.internmc.facebook.com/intern/diff/D16840394)

suo committed:
[jit] Make torch.jit.Attribute work with PYTORCH_ENABLED=0

SsnL commented on test/test_jit_disabled.py:
```diff
@@ -0,0 +1,51 @@
+    """
+    def test_attribute(self):
+        with _jit_disabled():
+            out_disabled = subprocess.check_output([
```
Instead of hardcoding this in every test method, can you use the `--subprocess` to run each method in subprocess 
https://github.com/pytorch/pytorch/blob/c002ede1075d05ab82e1d50fcc5f94ec1e0d95a9/test/common_utils.py#L50
and modify `run_test.py` to add a special handler for `test_jit_disabled` similar to
https://github.com/pytorch/pytorch/blob/c002ede1075d05ab82e1d50fcc5f94ec1e0d95a9/test/run_test.py#L113-L114?

SsnL replied:
And the `setUp` method in `test_jit_disabled.py` should assert that the env flag is correctly set and pytorch jit is disabled.

driazati replied:
The environment variable has to be set before the test is run since it's checked when `import torch` happens, and it doesn't look like the existing subprocess stuff can handle that. Maybe something like a custom runner for this class like would work:

```python
class TestJitDisabled(JitTestCase):
    def run(self, result=None):
        with _jit_disabled():
			super(TestJitDisabled, self).run(result)
```

driazati commented on test/test_jit_disabled.py:
```diff
@@ -0,0 +1,51 @@
+print(s.x)
+"""
+
+class TestJitDisabled(unittest.TestCase):
```
Can you move https://github.com/pytorch/pytorch/blob/489cc46686e7e689e7130e78b73fe77a9edac5ed/test/test_jit.py#L2849 over here as well

zdevito reviewed (approved):
I have no opinion about how to handle the test. The code changes look fine.

suo committed:
Update on "[jit] Make torch.jit.Attribute work with PYTORCH_ENABLED=0"

suo committed:
Update on "[jit] Make torch.jit.Attribute work with PYTORCH_ENABLED=0"


Differential Revision: [D16840394](https://our.internmc.facebook.com/intern/diff/D16840394)

ShangxuanWu commented:
The title of this PR should be `PYTORCH_JIT=1` right?

driazati commented:
Yeah that’s correct
